packages/runner/injection/cookies.js

```diff
@@ -0,0 +1,75 @@
+/* global document */
+
+// document.cookie monkey-patching
+// -------------------------------
+// We monkey-patch document.cookie when in a cross-origin injection, because
+// document.cookie runs into cross-origin restrictions when the AUT is on
+// a different origin than top. The goal is to make it act like it would
+// if the user's app was run in top.
+//
+// The general strategy is:
+// - Keep the document.cookie value (`documentCookieValue`) available so
+//   the document.cookie getter can synchronously return it.
+// - Optimistically update that value when document.cookie is set, so that
+//   subsequent synchronous calls to get the value will work.
+// - On an interval, get the browser's cookies for the given domain, so that
+//   updates to the cookie jar (via http requests, cy.setCookie, etc) are
+//   reflected in the document.cookie value.
+export const patchDocumentCookie = (Cypress) => {
+  const setAutomationCookie = (toughCookie) => {
+    const { superDomain } = Cypress.Location.create(window.location.href)
+    const automationCookie = Cypress.Cookies.toughCookieToAutomationCookie(toughCookie, superDomain)
+
+    Cypress.automation('set:cookie', automationCookie)
+    .catch(() => {
+      // unlikely there will be errors, but ignore them in any case, since
```

Comment:
A debug log here would be great but I don't know if it would really provide anything since I believe this won't get put in stdout

Reply:
I agree on both counts, but since debug logs don't really output and I don't want to introduce any dependencies into the injection script (to keep it as small as possible), I'm leaning towards not using a debug log here or in the below catch block.